src/app/invest/invest.component.ts

```diff
@@ -64,25 +63,25 @@ export class InvestComponent {
       this.alreadyInvested$,
     ]).pipe(take(1),
       map(([campaign, balance, alreadyInvested]) => {
-        const campaignMin = Number(utils.formatEther(campaign.minInvestment))
-        const campaignMax = Number(utils.formatEther(campaign.maxInvestment))
+        const walletBalance = Number(formatEther(balance))
 
-        const campaignTokensBalance = Number(utils.formatEther(campaign.totalTokensBalance))
-        const campaignTokensSold = Number(utils.formatEther(campaign.totalTokensSold))
-        const campaignTokenPrice = TokenPrice.parse(campaign.tokenPrice.toNumber())
-        const campaignTokensAvailable = campaignTokensBalance - campaignTokensSold
-        const campaignAmountAvailable = campaignTokensAvailable * campaignTokenPrice
+        const campaignMin = Number(formatEther(campaign.minInvestment))
+        const campaignMax = Number(formatEther(campaign.maxInvestment))
 
-        const walletBalance = Number(utils.formatEther(balance))
+        const campaignStats = this.campaignService.stats(campaign)
 
         const min = alreadyInvested > 0 ? 0 : campaignMin
-        let max = campaignAmountAvailable < campaignMax ? campaignAmountAvailable : campaignMax
 
-        // ensure floor on 2 decimal points
-        max = Math.round(Math.floor(max * 100)) / 100
+        let userInvestGap = this.floorDecimals(campaignMax - alreadyInvested)
+        let max = this.floorDecimals(Math.min(userInvestGap, campaignStats.valueToInvest))
+
+        userInvestGap = Math.floor(userInvestGap * 100) / 100
```

Comment:
Why not use the function `floorDecimals`?

```suggestion
        userInvestGap = floorDecimals(userInvestGap)
```